File: app/src/main/java/com/i3ni/geeknum1/view/BaseTabView.java
package com.i3ni.geeknum1.view;


import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;

import com.i3ni.geeknum1.R;
import com.i3ni.geeknum1.bean.TabBean;

/**
 * Created by Mustang on 19/4/30.
 */

public abstract class BaseTabView extends LinearLayout {
    protected TabBean bean;


    public BaseTabView(Context context) {
        super(context);
    }

    public BaseTabView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public BaseTabView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setData(TabBean bean) {
        this.bean = bean;
        initView();
    }

    public abstract void initView();


}

File: app/src/main/java/com/i3ni/geeknum1/view/BaseTabView.kt
package com.i3ni.geeknum1.view


import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout

import com.i3ni.geeknum1.R
import com.i3ni.geeknum1.bean.TabBean

/**
 * Created by Mustang on 19/4/30.
 */

abstract class BaseTabView : LinearLayout {
    protected lateinit var bean: TabBean


    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    fun setData(bean: TabBean) {
        this.bean = bean
        initView()
    }

    abstract fun initView()


}

